File: Modul-4/Soal-4/PRAK404-2410817220030-RACHELWINAYUDA.py
def kalkulator():

    while True:
        print("\nPilih program")
        print("1. Penjumlahan")
        print("2. Pengurangan")
        print("3. Perkalian")
        print("4. Pembagian")
        print("5. Exit")
        pilihan = input("Masukkan Pilihan : ")

        if pilihan not in ['1', '2', '3', '4', '5']:
            print("Input anda salah, silahkan coba lagi")
            continue
        
        if pilihan == '5':
            print(f"Terimakasih, telah menggunakan kalkulator RACHELWINAYUDA")
            break

        nilai_pertama = float(input("Masukkan nilai pertama : "))
        nilai_kedua = float(input("Masukkan nilai kedua : "))

        if pilihan == '1':
            hasil = nilai_pertama + nilai_kedua
            print(f"Hasil Penjumlahan antara {nilai_pertama:.2f} dengan {nilai_kedua:.2f} adalah {hasil:.2f}")
        elif pilihan == '2':
            hasil = nilai_pertama - nilai_kedua
            print(f"Hasil Pengurangan antara {nilai_pertama:.2f} dengan {nilai_kedua:.2f} adalah {hasil:.2f}")
        elif pilihan == '3':
            hasil = nilai_pertama * nilai_kedua
            print(f"Hasil perkalian antara {nilai_pertama:.2f} dengan {nilai_kedua:.2f} adalah {hasil:.2f}")
        elif pilihan == '4':
            if nilai_kedua != 0:
                hasil = nilai_pertama / nilai_kedua
                print(f"Hasil Pembagian antara {nilai_pertama:.2f} dengan {nilai_kedua:.2f} adalah {hasil:.2f}")
            else:
                print("Error: Pembagian dengan nol tidak diperbolehkan")
                
kalkulator()

File: Modul-4/Soal-4/PRAK404-2410817220030-RACHELWINAYUDA.c
#include <stdio.h>

int main() {
    int pilihan;
    float nilai_pertama, nilai_kedua, hasil;

    while (1) {
        printf("\nPilih program\n");
        printf("1. Penjumlahan\n");
        printf("2. Pengurangan\n");
        printf("3. Perkalian\n");
        printf("4. Pembagian\n");
        printf("5. Exit\n");
        printf("Masukkan Pilihan : ");
        scanf("%d", &pilihan);

        if (pilihan < 1 || pilihan > 5) {
            printf("Input anda salah, silahkan coba lagi\n");
            continue;
        }

        if (pilihan == 5) {
            printf("Terimakasih, telah menggunakan kalkulator RACHELWINAYUDA\n\n");
            break;
        }

        printf("Masukkan nilai pertama : ");
        scanf("%f", &nilai_pertama);
        printf("Masukkan nilai kedua : ");
        scanf("%f", &nilai_kedua);

    switch (pilihan) {
        case 1:
            hasil = nilai_pertama + nilai_kedua;
            printf("Hasil Penjumlahan antara %.2f dengan %.2f adalah %.2f\n", nilai_pertama, nilai_kedua, hasil);
            break;
        case 2:
            hasil = nilai_pertama - nilai_kedua;
            printf("Hasil Pengurangan antara %.2f dengan %.2f adalah %.2f\n", nilai_pertama, nilai_kedua, hasil);
            break;
        case 3:
            hasil = nilai_pertama * nilai_kedua;
            printf("Hasil Perkalian antara %.2f dengan %.2f adalah %.2f\n", nilai_pertama, nilai_kedua, hasil);
            break;
        case 4:
            if (nilai_kedua == 0) {
                printf("Error: Pembagian dengan nol tidak diperbolehkan\n");
            } else {
                hasil = nilai_pertama / nilai_kedua;
                printf("Hasil Pembagian antara %.2f dengan %.2f adalah %.2f\n", nilai_pertama, nilai_kedua, hasil);
            }
            break;
        }
    }
    return 0;
}